test/Orleans.ProtobufNet.Tests/ProtobufNetTests.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Orleans.Serialization.ProtobufNet;
+using System;
+using System.Reflection;
+using TestExtensions;
+using Orleans.Runtime.Configuration;
+using Xunit;
+using System.IO;
+using FluentAssertions;
+
+namespace Orleans.ProtobufNet.Tests
+{
+    public class ProtobufNetTests
+    {
+        private readonly SerializationTestEnvironment environment;
```

Comment:
suggest turn `GoogleUtils.Tests.Serialization.ProtobufSerializationTests` tests into a test runner. So that both ProtobugSerializer and ProtobufNetSerializer can run the same set of tests, by using the same test runner. 

Also with the test runner, we can slowly adding tests to the test runner to increase coverage and all external serializer will benefit at the same time. 

But bottom line, the tests here only copied two of the tests from  `GoogleUtils.Tests.Serialization.ProtobufSerializationTests`, we should at least have them all?

Reply:
What do you mean by a test-runner? They use different classes and different libraries/methods, so I'm having a hard time seeing how we can make the tests cover both Google Protobuf and protobuf-net. Any ideas? I can copy over all the other tests as well no problem, but I understand that ideally they would be the same tests

Reply:
Or did you mean to simply have the protobuf-net tests copied into the same tests as the GoogleUtils one?
As in

```csharp
public void ProtobufSerializationTest_3_ProtoSerialization()
{
       var input = CreateAddressBook();
       var input2 = CreatePerson();
       ..... // Test both types and libraries
}
```

Reply:
a test runner is a set of tests which can be applied to different implementation of the same component/interface. For example, `GrainPersistenceTestsRunner` , is used in AdoNet persistence tests and Azure persistence tests. 

Yes I understand those two serializer uses different libraries, but they are both `IExternalSerializer`, is it not? the test runner should be `ExternalSerializerTestRunner` which can be applied to all IExternalSerializer implementations. Does this make sense? 

I suggested copy tests over basically to increase the test coverage for this serializer when you don't have time to make a test runner.  Since they are both IExternalSerializer so they should pass the same set of tests. Is it not the case?

Reply:
Ahh, thanks! I'll give it a shot later today